store/tikv/client.go

```diff
@@ -506,6 +514,23 @@ func (a *connArray) batchSendLoop(cfg config.TiKVClient) {
 	}
 }
 
+// removeCanceledRequests removes canceled requests before sending.
+func removeCanceledRequests(
```

Comment:
How about write the logic in `fetchPendingRequest` to reduce allocaction?

Reply:
It's possible that after wait for tikv overload, some of the requests canceled.
This way removes more canceled requests.